models/kernels/EtPriestleyTaylorMethodEx.hpp

```diff
@@ -0,0 +1,263 @@
+//    updated method similar to the one presented in Bras, R.L. Hydrology.  Requires value of the Linke atmospheric
+//    turbidity factor, which varies from 2 for clear mountain air to 5 for smoggy air.  According to Hove & Manyumbu
+//    2012, who calculated values over Zimbabwe that varied from 2.14 to 3.71.  Other values exist in the literature.
+//    TODO: This turbidity factor could be calculated from satellite obs. or maybe NOAA already does this?
+//
+// All radiation calculations needed for 1, 3, 4, and 5 require net radiation calculations at the land surface.
+// the net radiation is calculated using a, c, d, e, f, g, j, plus the Linke turbidity factor, which can be estimate
+// from satellite observations.
+//
+
+// NOTE THE VALUE OF evapotranspiration_params.zero_plane_displacement_height COMES FROM LAND COVER DATA.
+// Taken from:    https://websites.pmc.ucsc.edu/~jnoble/wind/extrap/
+//
+//Roughness Roughness  Landscape Type
+// Class    Length (m)	
+//----------------------------------------------------------------------------------------------------------------
+// 0         0.0002     Smooth water surface
+// 0.2       0.0005     Inlet water
+// 0.5       0.0024     Completely open terrain, smooth surface, e.g. concrete runways in airports, mowed grass, etc.
+// 1         0.03       Open agricultural area without fences and hedgerows and very scattered buildings. Only softly
+//                        rounded hills
+// 1.5       0.055      Agricultural land with some houses and 8 metre tall sheltering hedgerows with a distance of 
+//                        approximately 1250 metres
+// 2         0.1        Agricultural land with some houses and 8 metre tall sheltering hedgerows with a distance of 
+//                        approximately 500 metres
+// 2.5       0.2        Agricultural land with many houses, shrubs and plants, or 8 metre tall sheltering hedgerows 
+//                        with a distance of approximately 250 metres
+// 3         0.4        Villages, small towns, agricultural land with many or tall sheltering hedgerows, forests 
+//                        and very rough and uneven terrain
+// 3.5       0.8        Larger cities with tall buildings
+// 4         1.6        Very large cities with tall buildings and skyscrapers
+// Roughness definitions according to the European Wind Atlas. 
+//
+// According to the UN FAO Penman-Monteith example here: 
+//        http://www.fao.org/3/X0490E/x0490e06.htm#aerodynamic%20resistance%20(ra)
+// the zero plane roughness length,"d" can be approximated as 2/3 of the vegetation height (H): d=2/3*H.
+// the momentum roughness height "zom" can be estimated as 0.123*H.
+// the heat transfer roughness height "zoh" can be approximated as 0.1 * zom.
+//-----------------------------------------------------------------------------------------------------------------
+
+double priestley_taylor_method_realization()
```

Comment:
the "realization" term is miss-represented.  The original intent of "improve realization handling of ET" was aimed at improving the calling of ET functions from a model formulation so that we weren't doing some funny business of casting `void *` functions and assuming the correct function and call signature.  These ET functions should be just that, functions with a unified "look and feel" if you will.  Similarly, I would make a sub directory under `models/kernels` to hold these various modules `models/kernels/evapo_transpiration/` is probably a good place to start.  Might also consider wrapping them all under an ET namespace.